### ISSUE TITLE
iput with force flag does not delete existing oid from WOS archive

### ISSUE THREAD

**Issue by richard-mansfield** [irods/irods_resource_plugin_wos#1574]
In this case, we use a Replicated Object Assure policy names "irodsTest" thereby we expect 2 objects replicas in two WOS zones, one per zone. Putting a file increases the number of WOS object replicas by 2. Overwriting using "iput -f" force flag does not delete the old object.  Object replica count is now 4 and the previous object can be retrieved using the WOS object ID (oid).

Issue reproducer:

```
ires:~> curl -s "http://admin:admin@wosNode01:8088/mgmt/policy_info" | \
python -mjson.tool | grep -A3 name
        "name": "irodsTest",
        "notify": 0,
        "objAssure": 1,
        "objects": 0,

ires:~> ilsresc
archiveTest:compound
├── cacheTest
└── wosTest:wos

ires:~> echo oldObject > archiveTest.txt

ires:~> iput --purgec archiveTest.txt

ires:~> curl -s "http://admin:admin@wosNode01:8088/mgmt/policy_info" | \
python -mjson.tool | grep -A3 name
        "name": "irodsTest",
        "notify": 0,
        "objAssure": 1,
        "objects": 2,

ires:~> ils -L
/demoZone/home/rods:
rods   1   archiveTest;wosTest   26 2016-05-05.17:01 & archiveTest.txt
        generic    1A0njnSRBPpB4dBl_tXLVgsm0DQ7oeF91pZAMU-M

ires:~> echo newObject > archiveTest.txt

ires:~> iput -f --purgec archiveTest.txt

ires:~> curl -s "http://admin:admin@wosNode01:8088/mgmt/policy_info" | \
python -mjson.tool | grep -A3 name
        "name": "irodsTest",
        "notify": 0,
        "objAssure": 1,
        "objects": 4,

ires:~> ils -L
/demoZone/home/rods:
rods   1   archiveTest;wosTest   26 2016-05-05.17:02 & archiveTest.txt
        generic    eDfMMZo6CkG_C2CuCkugryuteNc6EtOX7P-sL4eL

ires:~> curl -vs -X GET -o out.txt  -H "x-ddn-oid:1A0njnSRBPpB4dBl_tXLVgsm0DQ7oeF91pZAMU-M" http://wosNode01/cmd/get

ires:~> cat out.txt
oldObject

ires:~> irm -f archiveTest.txt

ires:~> curl -s "http://admin:admin@wosNode01:8088/mgmt/policy_info" | \
python -mjson.tool | grep -A3 name
        "name": "irodsTest",
        "notify": 0,
        "objAssure": 1,
        "objects": 2,

```

**Comment by trel** [comment]
Without looking at any code, it seems the 'force' flag is not propagating into the WOS API appropriately.

Does the `objects` count still go to `4` without using `--purgec`?

**Comment by richard-mansfield** [comment]
Yes, the count still goes to 4 without using the --purgec flag.

**Comment by jasoncoposky** [comment]
We will reproduce this and roll it into the next release in support of iRODS 4.2.

**Comment by francescogiannoccaro** [comment]
Hi Jason,
has the fix made in the 4.2 release?

**Comment by francescogiannoccaro** [comment]
a temporary work around would be using a acPreProc rule for creating an Object.
There is acPostProcForDataObjWrite, is there an equivalent PreProc rule?
In this way maybe we could prevent users from user the force flag and write to the screen an error message?

**Comment by trel** [comment]
We have not released iRODS 4.2 yet.  The WOS plugin will be released separately once it can be tested against the 4.2 core.

There is only a static policy enforcement point for `acPreprocForDataObjOpen`.

There is a dynamic policy enforcement point you could use... `pep_resource_write_pre`.
